src/middleware/auth.ts

```diff
@@ -0,0 +1,30 @@
+import { Request, Response, NextFunction } from "express";
+import { verifyToken } from "../auth/auth.services";
+import { AuthUser } from "../auth/auth.types";
+
+export const auth = (
+  req: Request & AuthUser,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const { authorization } = req.headers;
+
+    if (!authorization) {
+      throw new Error('Your session has expired, please log in again');
+    }
+
+    const [, token] = authorization.split(' ');
+
+    if (!token) {
+      throw new Error('Your session has expired, please log in again');
+    }
+
+    const { id } = verifyToken(token) as { id: string };
```

Comment:
`{ id: string }` could be a type or interface.

Reply:
Resolved by creating an alias, solves the id issue from above too